libraries/standard/mqtt/utest/mqtt_state_utest.c

```diff
@@ -68,45 +69,193 @@ static void fillRecord( MQTTPubAckInfo_t * records,
     }
 }
 
+static void validateRecordAt( MQTTPubAckInfo_t * records,
+                              size_t index,
+                              uint16_t packetId,
+                              MQTTQoS_t qos,
+                              MQTTPublishState_t state )
+{
+    TEST_ASSERT_EQUAL( packetId, records[ index ].packetId );
+    TEST_ASSERT_EQUAL( qos, records[ index ].qos );
+    TEST_ASSERT_EQUAL( state, records[ index ].publishState );
+}
+
 /* ========================================================================== */
 
 void test_MQTT_ReserveState( void )
 {
     MQTTContext_t mqttContext = { 0 };
     MQTTStatus_t status;
+    const uint16_t PACKET_ID = 1;
+    const uint16_t PACKET_ID2 = 2;
+    const uint16_t PACKET_ID3 = 3;
+    const size_t index = 0;
```

Comment:
I don't think it's worth declaring this const as 0. I think it's more readable to just leave 0, since that immediately tells me we're testing the first element